Toolchain/BuildFuseExt2.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/sh
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+die() {
+    echo "die: $*"
+    exit 1
+}
+
+if [[ "$OSTYPE" != "darwin"* ]]; then
+    die "This script makes sense to be run only on macOS"
+fi
+export PATH=/opt/gnu/bin:$PATH
```

Comment:
I think it's an established convention that building Serenity does not impact your system, i.e.all the scripts don't touch anything outside of the Serenity root.

Reply:
Well, I am a big fan of self contained systems and softwares,  but on macOS the quickest way I've found to create and mount an ext2 volume is to install fuse-ext2, that installs as a "system" FUSE driver/backend.
I think you can see this as a "prerequisite", exactly like the apt-get for libraries and GCC8 that are listed on ReadMe in the Linux session.

Probably we can redirect most of these scripts to build and install things locally, but I am not sure if one can get entire fuse-ext2 installed without putting its libraries into a "system" folder.
If this behavior is a problem we can put just a stub here with a message linking to the script living in some other repo.

Something that would be cool to do is to be able compiling and using use the EXT2 code outside of Kernel, to build the image from scratch with an internal helper tool without external dependencies. I am not sure how difficult would be to do it, I would need to look at the code.
That would make it easily possible to build and run Serenity on Windows too!

Reply:
> I think you can see this as a "prerequisite", exactly like the apt-get for libraries and GCC8 that are listed on ReadMe in the Linux session.

Sure; but the difference is that apt is already *the* right way to install packages (on Debian variants), it does install them to the right directories and uses the right settings / build options.

You're building something yourself, and you're hardcoding stuff like paths; I may want the software in say `/usr/local` instead of `/opt/gnu`. I pick a Linux distro based on how I like these things organized, then I install gcc dependencies in accordance with my distro's policies; but this script basically hardcodes its own policy, which would be okay if it was Serenity-specific and wouldn't touch system-wide stuff. Maybe I do already have some of the necessary software installed, after all, what would happen then?

So I would like this script to at least asks the user for their preferred PATH/prefix, maybe as an optional argument to the script, maybe as an interactive thing you have to confirm.

Besides, isn't this stuff (automake, autoconf, libtool, ext2fuse, ext2fsprogs) already available in Homebrew?

Reply:
I have tried installing as much as possible from brew.
Unfortunately `fuse-ext2` brew formula is not available, and it will still need the custom install script.